src/CommunityToolkit.Maui.Core/Views/Popup/MauiPopup.android.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Android.App;
+using Android.Content;
+using CommunityToolkit.Maui.Core;
+using Microsoft.Maui.Platform;
+using AView = Android.Views.View;
+
+namespace CommunityToolkit.Core.Views;
+
+/// <summary>
+/// The navite implementation of Popup control.
+/// </summary>
+public class MauiPopup : Dialog, IDialogInterfaceOnCancelListener
+{
+	readonly IMauiContext mauiContext;
+
+	/// <summary>
+	/// Constructor of <see cref="MauiPopup"/>.
+	/// </summary>
+	/// <param name="context">An instance of <see cref="Context"/>.</param>
+	/// <param name="mauiContext">An instace of <see cref="IMauiContext"/>.</param>
+	/// <exception cref="ArgumentNullException">If <paramref name="mauiContext"/> is null an exception will be thrown. </exception>
+	public MauiPopup(Context context, IMauiContext mauiContext)
+		: base(context)
+	{
+		this.mauiContext = mauiContext ?? throw new ArgumentNullException(nameof(mauiContext));
+	}
+
+	/// <summary>
+	/// An instace of the <see cref="IPopup"/>.
+	/// </summary>
+	public IPopup? VirtualView { get; private set; }
```

Comment:
Is it correct that we use VirtualView in NativeView?

Reply:
Yes. We need it here because we want to have access to values from the shared code